### PR TITLE
Database setup moongo

### PR DESCRIPTION
**Pull request by william000000**
- Setup Mongo DB
- Versioning our app
- Create User Tables
- Set our package manager to be yarn

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# mongo "mongodb+srv://cluster0-wojc5.mongodb.net/test"  --username user-one
+# mongodb://username:password@host:port/database?options...
+MONGODB_URI = mongodb+srv://user-one:password%21@cluster0-wojc5.mongodb.net/test?retryWrites=true&w=majority
```
Comment (urbanishimwe) [inline .env]: won't it be necessary to add this file(`.env`) in `.gitignore`

File: src/V1/database/mongoConnection.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const MongoClient = require('mongodb').MongoClient;
+const uri = "mongodb+srv://user-one:password%21@cluster0-wojc5.mongodb.net/test?retryWrites=true&w=majority";
```
Comment (urbanishimwe) [inline src/V1/database/mongoConnection.js]:
```suggestion
const uri = process.env.NODE_ENV === 'production' ? process.env.MONGODB_URI || process.env.DATABASE_URL_DEV";
```

**Review by urbanishimwe — CHANGES_REQUESTED** [review]
looks good but consider this comments

File: .env (hunk 1 above)
Comment (william000000) [reply .env]: Thanks for reminding me!

Comment (jo-rdan) [inline .env:3]: @william000000, good job so far, but I think that these should not be here. In fact, the whole file should not be here. consider placing it in the `.gitignore` file and we're good to go!

Comment (jo-rdan) [inline .env]: @william000000, good job so far, but I think that these should not be here. In fact, the whole file should not be here. Consider placing this file in the `.gitignore` file and we're good to go!

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "editor.formatOnSave": true
+}
```
Comment (jo-rdan) [inline .vscode/settings.json]: Also consider placing these kind of files in the `.gitignore` file

**Review by jo-rdan — CHANGES_REQUESTED** [review]
Good work so far, attend to these request and we're good

**Comment by jo-rdan** [comment]
@william000000, If you can also squash your commits, it will be fine

File: .travis.yml [hunk 4]
```diff
@@ -1,8 +1,15 @@
 language: node_js
+os:
+  - linux
+  - osx
```
Comment (urbanishimwe) [inline .travis.yml]: why are we testing in two different OSes

**Review by jo-rdan — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]
LGTM

**Pull request merged**